src/testdrive/src/action/s3.rs

```diff
@@ -140,7 +140,10 @@ pub struct AddBucketNotifications {
     sqs_validation_timeout: Duration,
 }
 
-pub fn build_add_notifications(mut cmd: BuiltinCommand) -> Result<AddBucketNotifications, String> {
+pub fn build_add_notifications(
+    mut cmd: BuiltinCommand,
+    state: &State,
```

Comment:
Hmm, passing state into an action constructor is a little weird. Can we rework so that `AddBucketNotifications::sqs_validation_timeout` is an `Option<Duration>`, and the `unwrap_or(state.default_timeout)`  happens in the `redo` function instead?